### DISCUSSION TITLE
Swipe navigation

### DISCUSSION THREAD

**Discussion by RossiRPI** [HASwitchPlate/openHASP#133] [Ideas]
Would it be possible to do swiping gestures for page navigation instead of pressing a next/back button?

**Comment by fvanroie** [comment]
Somehting like this?

![9IAkyyR4UP](https://user-images.githubusercontent.com/15969459/114113145-456f0f00-98de-11eb-9d1e-37ebb74a8c3d.gif)

**Reply by RossiRPI** [reply]
Yes, that's it! Is this currently supported in 0.4.1? I haven't seen it listed anywhere.

**Reply by fvanroie** [reply]
This is just a quick proto-typing to see if it could work. It's not pushed yet.

**Reply by RossiRPI** [reply]
It looks great, just needs to work with a rotated screen as well.

**Reply by fvanroie** [reply]
It is included in the [latest builds](https://github.com/HASwitchPlate/openHASP/actions) for testing.
If it works well without issues, we can keep it.

**Reply by dgomes** [reply]
works great (as long as you don't swipe on top of a button 😬 )

**Reply by fvanroie** [reply]
I figured this feature needed some real-world testing first.
It probably makes sense to not to accept gestures from clickable objects...

**Reply by fvanroie** [reply]
I reduced the false positives by implementing these changes:

- Objects don't propagate swipe events to the page by _default_ anymore. Do we need to make this configurable?
- Page swipes don't activate objects on the new page immediately: You need to release the plate to finish the swipe action.

**Comment by fvanroie** [comment]
Due to possible adverse effects, I added a configurable `swipe` property with values 0/1.
This disables or enables the swipe gestures on *any* object (or a page using p1b0.swipe) allowing for a more granular approach to swiping.

**Reply by nagyrobi** [reply]
What's the default state?
Edit: it's off as it looks like, and that's great.

**Comment by fvanroie** [comment]
Since `v0.5.0` there is `swipe` parameter that you can enable on an object to enable swiping for page change.

**Reply by nagyrobi** [reply]
https://haswitchplate.github.io/openHASP-docs/0.5/#objects/#swipe
